### PR TITLE
Create Integrations for each new service

### PR DESCRIPTION
**Pull request by wshearn**
Create Integrations for each service created and use the integration key inside the syncset

/cc @lisa 
/cc @jewzaam

### CODE REVIEW

File: pkg/pagerduty/service.go [hunk 1]
```diff
@@ -108,22 +108,44 @@ func (data *Data) ParsePDConfig(osc client.Client) error {
...
-		return "", errors.New("Multiple services returned from PagerDuty")
+		return pdApi.Service{}, errors.New("Multiple services returned from PagerDuty")
 	}
 
-	return services.Services[0].ID, nil
+	return services.Services[0], nil
+}
+
+// GetIntegrationKey searches the PD API for an already existing service and returns the first integration key
+func (data *Data) GetIntegrationKey() (string, error) {
```
Comment (lisa) [inline pkg/pagerduty/service.go]: What does accessing the first integration key get us? And if there's more than one, what does that mean; does this method operate on a specific type of integration (eg DeadManSnitch) only?

File: pkg/pagerduty/service.go [hunk 2]
```diff
@@ -141,25 +163,37 @@ func (data *Data) CreateService() (string, error) {
...
 	}
 
 	newSvc, err := client.CreateService(clusterService)
 	if err != nil {
 		return "", err
 	}
 
-	return newSvc.ID, nil
+	clusterIntegration := pdApi.Integration{
+		Name: "Test",
```
Comment (lisa) [inline pkg/pagerduty/service.go]: Surely this isn't intended? Or is this PR a WIP?

File: pkg/pagerduty/service.go (hunk 1 above)
Comment (wshearn) [reply pkg/pagerduty/service.go]: DeadManSnitch will have it's own service. The problem is that integrations can have the same name, so we don't know if it is the proper one, if we throw an error I plan on adding metrics that track the error count so if we detect an issue we can go look. https://i.imgur.com/f0pcfGM.jpg

File: pkg/pagerduty/service.go (hunk 2 above)
Comment (wshearn) [reply pkg/pagerduty/service.go]: I had left that there while I try and figure out a better name, no idea what to call it.

File: pkg/controller/clusterdeployment/clusterdeployment_deleted.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ import (
 func (r *ReconcileClusterDeployment) handleDelete(request reconcile.Request, instance *hivev1.ClusterDeployment) (reconcile.Result, error) {
 	r.reqLogger.Info("Deleting syncset")
 
-	ssName := fmt.Sprintf("%v-pd-sync", instance.Name)
+	ssName := instance.Name + "-pd-sync"
```
Comment (jharrington22) [inline pkg/controller/clusterdeployment/clusterdeployment_deleted.go]: Why switch this?

Comment (wshearn) [reply pkg/controller/clusterdeployment/clusterdeployment_deleted.go]: After reading it all over it was just easier for me to read

File: pkg/kube/configmap.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+// limitations under the License.
+
+package kube
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GenerateConfigMap returns a syncset that can be created with the oc client
```
Comment (lisa) [inline pkg/kube/configmap.go]: This comment doesn't match up with the function. The function is returning a `*corev1.ConfigMap` but the comment seems to suggest a SyncSet would be returned.

**Pull request merged**